packages/library/src/commonMain/kotlin/io/realm/RealmList.kt

```diff
@@ -16,232 +16,31 @@
- *
- * [AbstractMutableList] provides enough default implementations on which we can rely. It can also
- * be used as a delegate since it implements [MutableList].
- */
-private class ManagedListDelegate<E>(
-    override val listPtr: NativePointer,
-    private val metadata: RealmList.OperatorMetadata
-) : AbstractMutableList<E>(), RealmListApi<E> {
-
-    private val operator = RealmList.Operator<E>(metadata)
-
-    override val size: Int
-        get() {
-            metadata.realm.checkClosed()
-            return RealmInterop.realm_list_size(listPtr).toInt()
-        }
-
-    override fun get(index: Int): E {
-        metadata.realm.checkClosed()
-        return operator.convert(RealmInterop.realm_list_get(listPtr, index.toLong()))
-    }
-
-    override fun add(index: Int, element: E) {
-        metadata.realm.checkClosed()
-        RealmInterop.realm_list_add(
-            listPtr,
-            index.toLong(),
-            copyToRealm(metadata.mediator, metadata.realm, element)
-        )
-    }
-
-    // FIXME bug in AbstractMutableList.addAll native implementation:
-    //  https://youtrack.jetbrains.com/issue/KT-47211
-    //  Remove this method once the native implementation has a check for valid index
-    override fun addAll(index: Int, elements: Collection<E>): Boolean {
-        metadata.realm.checkClosed()
-        rangeCheckForAdd(index)
-        return super.addAll(index, elements)
-    }
-
-    override fun clear() {
-        metadata.realm.checkClosed()
-        RealmInterop.realm_list_clear(listPtr)
-    }
-
-    override fun removeAt(index: Int): E = get(index).also {
-        metadata.realm.checkClosed()
-        RealmInterop.realm_list_erase(listPtr, index.toLong())
-    }
-
-    override fun set(index: Int, element: E): E {
-        metadata.realm.checkClosed()
-        return operator.convert(
-            RealmInterop.realm_list_set(
-                listPtr,
-                index.toLong(),
-                copyToRealm(metadata.mediator, metadata.realm, element)
-            )
-        )
-    }
-
-    override fun observe(list: RealmList<E>): Flow<RealmList<E>> {
-        metadata.realm.checkClosed()
-        return metadata.realm.owner.registerListObserver(list)
-    }
-
-    override fun freeze(frozenRealm: RealmReference): RealmList<E> {
-        val frozenList = RealmInterop.realm_list_freeze(listPtr, frozenRealm.dbPointer)
-        return RealmList(frozenList, metadata.copy(realm = frozenRealm))
-    }
-
-    override fun thaw(liveRealm: RealmReference): RealmList<E> {
-        val liveList = RealmInterop.realm_list_thaw(listPtr, liveRealm.dbPointer)
-        return RealmList(liveList, metadata.copy(realm = liveRealm))
-    }
-
-    private fun rangeCheckForAdd(index: Int) {
-        if (index < 0 || index > size) {
-            throw IndexOutOfBoundsException("Index: '$index', Size: '$size'")
-        }
-    }
-}
+fun <T> realmListOf(vararg elements: T): RealmList<T> =
```

Comment:
Should we also have an override that takes other collections?

Reply:
I added a `Iterable<T>.toRealmList()` extension method following the conventions from Kotlin.